src/vs/workbench/parts/terminal/electron-browser/terminal.contribution.ts

```diff
@@ -105,6 +105,11 @@ configurationRegistry.registerConfiguration({
 			'type': 'number',
 			'default': 1.2
 		},
+		'terminal.integrated.fontWeight': {
+			'type': 'string',
+			'description': nls.localize('terminal.integrated.fontWeight', "Customizes font weight in terminal."),
```

Comment:
This needs to be something like `enableBold` like how gnome-terminal does it, otherwise this setting will override the terminal's regular bold/normal formatting with what's in this setting.

The logic should then be added as an event listener on the `IConfigurationService` so it's picked up when the setting is changed, something like this:

```
set class name 'disable-bold' if `enableBold` setting is not set
```

I use `disable-bold` as the class name here because enableBold will be the standard.